### DISCUSSION TITLE
Hardware exhibition

### DISCUSSION THREAD

**Discussion by Renber78** [sle118/squeezelite-esp32#305] [Show and tell]
I suggest that you post photos as well as a small description of your completed projects here.
Given the many possibilities that we can achieve, it is very interesting to share our creations!

**Comment by Renber78** [comment]
Here is one of my squeeze ESP32s which is an addon to my living room amplifier.

![PXL_20230919_063533231 1](https://github.com/sle118/squeezelite-esp32/assets/68775667/fe956e60-4dae-4b64-a7ba-6d501c3e53b5)
![PXL_20230919_063843137 1](https://github.com/sle118/squeezelite-esp32/assets/68775667/34dc7041-f191-40f6-953e-b4cab58736c7)

- Made in a black anodized aluminum case.
- Driven by an ESP32-WROVER-B on a homemade PCB.
- The front panel in blue plexiglass hides an IPS ST7735 160x80px screen, an IR receiver TSOP4838 and a touch switch TTP223 (Power)
- The output is analog (DAC: PCM5102), it is not amplified.
- For power, it is a USB cable
- The remote control is from an old Philips TV (Combi TV/VCR) which fits perfectly (Code RC5)
- It has an external Wifi/BT antenna

**Comment by dfrein** [comment]
As far as I remember the "xbox one media remote" should also work, and you can easily get cheap alternatives at ebay. 
[XBoxOne_Remote.zip](https://github.com/sle118/squeezelite-esp32/files/12790834/XBoxOne_Remote.zip)

**Comment by haldi4803** [comment]
Using the [Louder ESP Board](https://github.com/sonocotta/esp32-audio-dock/tree/main)  bought on Tindie for few bucks has a TAS5805M Amplifier that does 2x23W.
Enough for most small satellite Speakers or 4" bookshelf speakers.

As long as it's anything decent and not such crappy things as those Cambridge Audio Mini Minx 20 it sounds quite decent.
![20231013_192936](https://github.com/sle118/squeezelite-esp32/assets/17104473/3f49cb1c-c735-4831-90a3-655fe0cf3134)

Getting a few more board and other speakers I'm one step closer to the dream of a Multi room Audio Setup.

The next thing would be a Willow Setup. (And microphones on all those Speakers) for Voice control.
https://heywillow.io

Edit:
Also got some decent Speakers.
![20231105_224707](https://github.com/sle118/squeezelite-esp32/assets/17104473/d6f466a5-28a6-4630-ad8b-c891f86d885b)

Just had to cover all the LEDs because it's a 24/7 device in my bedroom.

**Comment by holefrog** [comment]
I have a ESP32-A1S, and I found a $3 box in Dollarama. It would be enough to hold all the components.

- ESP32-A1S
- Rotary Encoder
- IR Receiver
- ST7789 TFT Display
![PXL_20231104_224722053](https://github.com/sle118/squeezelite-esp32/assets/17289895/9bee5083-bdac-476f-ad33-a6f1e5e68b12)
![PXL_20231104_220544718](https://github.com/sle118/squeezelite-esp32/assets/17289895/8e48235c-9554-43d6-a618-2f132e4f80dc)

**Reply by Androssboy** [reply]
Hello holefrog!
I have +\- the same hardware configuration.
Have you managed to implement IR control?
I installed GPIO in NVS, activated control in LMS plugins and edited the button codes in the Slim_Devices_Remote.ir file
The button codes were read using Arduino.
I suspect there is some secret that I don't know about. I have already read and studied many topics on the forums, but I am at a loss...
Please share your experience.

**Reply by holefrog** [reply]
Hi there,

My IR control works well. However, some of the GPIOs are behaving strangely. I have tried many combinations to make it work."

Below is my configuration:
```
{
  "a2dp_ctmt": "1000",
  "a2dp_ctrld": "500",
  "a2dp_dev_name": "ESP32-A1S",
  "a2dp_sink_name": "SMSL BT4.2",
  "a2dp_spin": "0000",
  "actrls_config": "buttons",
  "airplay_name": "ESP32-A1S",
  "airplay_port": "5000",
  "ap_channel": "1",
  "ap_ip_address": "192.168.4.1",
  "ap_ip_gateway": "192.168.4.1",
  "ap_ip_netmask": "255.255.255.0",
  "ap_pwd": "squeezelite",
  "ap_ssid": "ESP32-A1S",
  "autoexec": "1",
  "autoexec1": "squeezelite  -o i2s  -s  192.168.50.210:3483 -b  500:2000 -C  30 -d  all=info  -n \"ESP32-A1S\"",
  "bat_config": "",
  "board_model": "ESP32-A1s",
  "bt_name": "ESP32-A1S",
  "bt_sink_pin": "1234",
  "bt_sink_volume": "127",
  "buttons": " [{\"gpio\":13,\"pull\":true,\"normal\":{\"pressed\":\"ACTRLS_TOGGLE\"}}]",
  "bypass_wm": "0",
  "cspot_config": "{\t\"deviceName\":\t\"ESP32-A1S\",\t\"bitrate\":\t160,\t\"zeroConf\":\t1}",
  "dac_config": "model=ES8388,bck=27,ws=25,do=26,sda=33,scl=32,di=35,i2c=16",
  "dac_controlset": "",
  "dhcp_tmout": "8",
  "display_config": "SPI,width=320,height=240,back=19,rotate,VFlip,driver=ST7789",
  "enable_airplay": "1",
  "enable_bt_sink": "1",
  "enable_cspot": "1",
  "equalizer": "0,0,0,0,0,0,0,0,0,0",
  "eth_config": "",
  "ethtmout": "8",
  "gpio_exp_config": "",
  "host_name": "ESP32-A1S",
  "i2c_config": "",
  "jack_mutes_amp": "y",
  "led_brightness": "",
  "led_vu_config": "",
  "lms_ctrls_raw": "n",
  "loudness": "0",
  "metadata_config": "",
  "model_config": "ESP32A1S V2.2+ variant 1 (ES8388)",
  "ota_erase_blk": "249856",
  "ota_prio": "6",
  "ota_stack": "10240",
  "pollmin": "15",
  "pollmx": "600",
  "rel_api": "https://api.github.com/repos/sle118/squeezelite-esp32/releases",
  "release_url": "https://github.com/sle118/squeezelite-esp32/releases",
  "rotary_config": "A=3,B=22,SW=15,volume",
  "set_GPIO": "1=ir,  21=amp, 22=green:0, 39=jack:0",
  "sleep_config": "",
  "spdif_config": "",
  "spi_config": "mosi=23,clk=5,dc=18,port=1",
  "stats": "n",
  "target": "",
  "telnet_block": "500",
  "telnet_buffer": "40000",
  "telnet_enable": "",
  "wifi_smode": "f"
}
```

After flashing the Squeezelite firmware , have you checked the UART output while pressing any button on the remote?

Furthermore,  you do not need to modify the  Slim_Devices_Remote.ir , you can add another xxx.ir file which includes the codes in the same directory of  Slim_Devices_Remote.ir.  After that, you will see your remote control in LMS.
![ir](https://github.com/sle118/squeezelite-esp32/assets/17289895/e62b6398-3419-4caf-ac90-00d5035a238c)

**Reply by Androssboy** [reply]
Hello dear holefrog !
Thank you very much for your answer and for the NVS parameters (almost completely the same)!
I also tried many GPIOs, but with no result.
I am attaching a screenshot of the UART output and the configuration/boot log. Would it be difficult for you to take a look? When I press any button on the remote control, error messages are visible in the UART output window. I don't understand why I'm getting these errors?
![Errors](https://github.com/sle118/squeezelite-esp32/assets/103333832/054f3bb4-4a12-41e3-8f2e-bddbf2157d24)

**UART output**
`rst:0x1 (POWERON_RESET),boot:0x1f (SPI_FAST_FLASH_BOOT)
configsip: 0, SPIWP:0xee
clk_drv:0x00,q_drv:0x00,d_drv:0x00,cs0_drv:0x00,hd_drv:0x00,wp_drv:0x00
mode:DIO, clock div:1
load:0x3fff0030,len:6424
load:0x40078000,len:15632
load:0x40080400,len:4348
entry 0x40080688
I (27) boot: ESP-IDF v4.3.5-dirty 2nd stage bootloader
I (27) boot: compile time 06:35:58
I (27) boot: chip revision: v1.0
I (30) qio_mode: Enabling QIO for flash chip WinBond
I (36) boot.esp32: SPI Speed      : 80MHz
I (41) boot.esp32: SPI Mode       : QIO
I (45) boot.esp32: SPI Flash Size : 4MB
I (50) boot: Enabling RNG early entropy source...
I (55) boot: Partition Table:
I (59) boot: ## Label            Usage          Type ST Offset   Length
I (66) boot:  0 nvs              WiFi data        01 02 00009000 00004000
I (74) boot:  1 otadata          OTA data         01 00 0000d000 00002000
I (81) boot:  2 phy_init         RF data          01 01 0000f000 00001000
I (88) boot:  3 recovery         factory app      00 00 00010000 00140000
I (96) boot:  4 ota_0            OTA app          00 10 00150000 002a0000
I (103) boot:  5 settings         WiFi data        01 02 003f0000 00010000
I (111) boot: End of partition table
I (115) esp_image: segment 0: paddr=00150020 vaddr=3f400020 size=aeae0h (715488) map
I (321) esp_image: segment 1: paddr=001feb08 vaddr=3ffbdb60 size=01510h (  5392) load
I (323) esp_image: segment 2: paddr=00200020 vaddr=400d0020 size=1b2e58h (1781336) map
I (817) esp_image: segment 3: paddr=003b2e80 vaddr=3ffbf070 size=0452ch ( 17708) load
I (823) esp_image: segment 4: paddr=003b73b4 vaddr=40080000 size=1e014h (122900) load
I (863) esp_image: segment 5: paddr=003d53d0 vaddr=400c0000 size=00068h (   104) load
I (878) boot: Loaded app from partition at offset 0x150000
I (878) boot: Disabling RNG early entropy source...
I (890) psram: This chip is ESP32-D0WD
I (890) spiram: Found 64MBit SPI RAM device
I (890) spiram: SPI RAM mode: flash 80m sram 80m
I (893) spiram: PSRAM initialized, cache is in low/high (2-core) mode.
I (901) cpu_start: Pro cpu up.
I (904) cpu_start: Starting app cpu, entry point is 0x400813fc
I (0) cpu_start: App cpu up.
I (1407) spiram: SPI SRAM memory test OK
I (1420) cpu_start: Pro cpu start user code
I (1420) cpu_start: cpu freq: 240000000
I (1420) cpu_start: Application information:
I (1423) cpu_start: Project name:     Squeezelite-ESP32
I (1429) cpu_start: App version:      I2S-4MFlash-1670
I (1435) cpu_start: Compile time:     Jan 28 2024 06:37:30
I (1441) cpu_start: ELF file SHA256:  0000000000000000...
I (1447) cpu_start: ESP-IDF:          v4.3.5-dirty
I (1452) cpu_start: Min chip rev:     v1.0
I (1457) cpu_start: Max chip rev:     v3.99
I (1462) cpu_start: Chip rev:         v1.0
I (1467) heap_init: Initializing. RAM available for dynamic allocation:
I (1474) heap_init: At 3FFAFF10 len 000000F0 (0 KiB): DRAM
I (1480) heap_init: At 3FFB6388 len 00001C78 (7 KiB): DRAM
I (1487) heap_init: At 3FFB9A20 len 00004108 (16 KiB): DRAM
I (1493) heap_init: At 3FFC6EC8 len 00019138 (100 KiB): DRAM
I (1499) heap_init: At 3FFE0440 len 00003AE0 (14 KiB): D/IRAM
I (1506) heap_init: At 3FFE4350 len 0001BCB0 (111 KiB): D/IRAM
I (1512) heap_init: At 4009E014 len 00001FEC (7 KiB): IRAM
I (1518) spiram: Adding pool of 4018K of external SPI memory to heap allocator
I (1527) spi_flash: detected chip: winbond
I (1531) spi_flash: flash io: qio
I (1538) cpu_start: Starting scheduler on PRO CPU.
I (0) cpu_start: Starting scheduler on APP CPU.
I (1546) spiram: Reserving pool of 64K of internal memory for DMA/internal allocations
I (1556) esp_app_main: System is booting from power on.
I (1566) esp_app_main: Reset reason is: 1
I (1566) esp_app_main: Reboot counter=1

I (1566) esp_app_main: Starting app_main
I (1576) nvs_utilities: Initializing flash nvs
I (1596) nvs_utilities: Initializing nvs partition settings
I (1636) esp_app_main: Setting up telnet.
I (1636) telnet: ***Redirecting log output to telnet
E (131506) rmt: RMT RX BUFFER FULL
E (131666) rmt: RMT RX BUFFER FULL
E (131926) rmt: RMT RX BUFFER FULL
E (133146) rmt: RMT RX BUFFER FULL
E (133196) rmt: RMT RX BUFFER FULL
E (133906) rmt: RMT RX BUFFER FULL
E (135166) rmt: RMT RX BUFFER FULL
E (135216) rmt: RMT RX BUFFER FULL
E (135726) rmt: RMT RX BUFFER FULL
E (135776) rmt: RMT RX BUFFER FULL
E (143306) rmt: RMT RX BUFFER FULL
`

**Reply by Androssboy** [reply]
Surprisingly, today, after several reboots, the reaction to pressing the remote control buttons in the terminal has changed!
Unfortunately, I don’t have any of the ones suggested in the thread about IR control. Now the question is: how can I get another of the remote controls I have to work? Do I need to somehow additionally convert the button codes so that the LMS begins to recognize them?
![Output](https://github.com/sle118/squeezelite-esp32/assets/103333832/c1032710-8b2a-4621-a357-4c4247436c43)

**Reply by philippe44** [reply]
LMS can have custom codes for IR, it's one of the config files. I don't remember which one exactly but even ChatGPT might be able to answer

**Reply by holefrog** [reply]
Hi, Androssboy 

I can understand your current feelings, just like mine several months ago. It looks like exactly the problem I had before.  My problem was:
```
I (38848) IR: can't decode IR signal of len 18 
I (38848) IR: 10 26 cd 8e 08 29 ef 83 f4 04 c6 83 02 05 b8 83  
I (38848) IR: 14 05 a5 83 ef 04 cb 83 04 05 b6 83 e7 04 ba 83 
 I (38868) IR: f6 04 c4 83 20 0e bb 83 90 12 a2 83 91 09 c5 83  
I (38868) IR: 0c 05 98 83 a9 09 ab 83 e7 04 d2 83 1e 05 85 83  
I (38878) IR: 8e 09 c8 83 08 05 00 80 
```

I am new to LMS and SqueezeLite too. Renber78 helped me a lot at IR. You can find it in the topic [https://github.com/sle118/squeezelite-esp32/discussions/330]. The cause of the wrong decoding is the GPIOs of IR. I moved my previous GPIO of IR to my current set and fixed the decoding problem. Just try to use my current setting for IR and test again. To isolate the cause, you may disconnect all other peripherals.

**Reply by philippe44** [reply]
What gpio do you use?

**Comment by alidaf** [comment]
I've just finished a new 3D printed SqueezeAMP case to complement my other designs. I designed the speakers, which are also 3D printed. The small ones were using some old drivers I found with no information on them but the Hecate 1s use some decent Mark Audio CHN-50 drivers. The new SqueezeAMP case is middle top in the first photo, middle in the second, and on it's own in the last few. The others are various early versions of basic WROVER dev boards with third party DACs or SPDIF boards, a couple of MAX9744 amplifiers with volume display and IR, a TPA3250 power amp and VU meters. My very first design was for PiCorePlayer in the second photo (with the full colour display). I'm also currently working on a dual led matrix spectrum analyser and a valve pre-amp.

The dial on the SqueezeAMP is actually a rotary encoder with 5 buttons built-in so there are a lot of configuration options for UI navigation.

I'll be posting the step files for the SqueezeAMP case on Printables once I've taken stock of the extra components needed. @philippe44 has also asked for a Li-ion battery option so I will be doing some further space optimisation to try and accommodate that request, which I may complete also before posting the files.

![SqueezeAMP1](https://github.com/sle118/squeezelite-esp32/assets/14184640/3b30210b-d98b-440e-96bb-f53f2bfc1346)
![SqueezeAMP2](https://github.com/sle118/squeezelite-esp32/assets/14184640/bc8a3530-a3cc-4201-9ed2-bd64e7a21907)
![SqueezeAMP3](https://github.com/sle118/squeezelite-esp32/assets/14184640/491aecdb-6f3a-47a7-bedc-9cb71f20a0e1)
![SqueezeAMP4](https://github.com/sle118/squeezelite-esp32/assets/14184640/e868d8ca-d1d6-49a3-a02b-a80c52c088e9)
![SqueezeAMP5](https://github.com/sle118/squeezelite-esp32/assets/14184640/059cef71-0e8b-42b0-8810-100d0975be67)

**Reply by sle118** [reply]
Very nice. Could you also post a close up of the rotary with integrated buttons? I'm intrigued.

**Reply by alidaf** [reply]
Here's the product [link](https://learn.adafruit.com/ano-rotary-encoder/overview) - I think it'll be better and more explanatory than my photography! The breakout is sold separately but they still remain reasonably cheap.
I've made it slightly recessed without affecting the performance. There is an inner ring that spins and the directional buttons are behind the outer ring, plus the usual rotary centre clicker. I came across it by chance after already ordering a load of buttons, which is typically ironic.

**Reply by bungh0l10** [reply]
Wow this is awesome!  @alidaf are you accepting orders for those cases by any chance ? ^^

**Reply by alidaf** [reply]
If you mean the SqueezeAmp case, then I'm just trying to source another cable for the connector for a SqueezeAMP board to build another but produce step by step photos and then post the build process on Printables like my previous efforts [here](https://www.printables.com/model/589907-squeezeboxen-case-v1-v2-coming-soon), at which point the whole thing will be open-sourced and then possible for anyone to print. I tend to build 2 of every project I make. The first is the prototype and the second is the more deliberate, methodical documentation. I've been trying to source a second cable for months but the postage is ridiculous unless I spend orders of magnitude more on other products that I don't need so the process has effectively stalled.

The main concern I have for selling a case would be tolerances with the displays (they are very variable) and you being able to source the same components yourself, but it is a possibility. I can supply Everything except the SqueezeAmp board and cables/wires, all fitted if needed but I wouldn't be keen on doing all the soldering and crimping again. Also, be aware that I had to remove the IR from the squeezeamp board and solder a new header in it's place. The display and sockets are from AliExpress, so easily sourced. The rotary dial can be bought from Mouser or Digikey but the optical pass-through was from a UK supplier so could be an issue if you are outside the UK. Adapting the design to an alternative wouldn't be a problem but getting hold of one myself could be, and that's the trap I fell into with the SqueezeAmp cable.

There are some other options at the link given earlier too. I'm about to add an optical output to the PCM5102 and ES9018 cases, at which point my SqueezeAmp case will become redundant and I could sell that (minus the SqueezeAmp board), since I only use the optical out on it. Once the second is built I'll probably sell that minus the board as well but the colour scheme is different (grey and blue).

I don't think it's possible to PM me here but can through Printables.

**Comment by zarss** [comment]
Not as neat as yours, but exactly what I was after. One is set as left, another as right channels.
![20240129_110337](https://github.com/sle118/squeezelite-esp32/assets/33965048/cec3b143-9ab3-4c6c-81a8-1b4c39563d06)

**Reply by alidaf** [reply]
Nice hacks. I'm always impressed with repurposing things like this. I'd like to do some real classic stuff - it's just finding the right skip before the vultures get there first.

**Comment by rjgrandy** [comment]
I gutted an echo 4th gen and replaced with: ESP32, PCM5102, Parts Express 320-608 small 2.1 Amplifier Board, and a buck converter. 

It works okay. The 2.1 board needs >7V to function properly, and can go up to 18V. So I fed 18V to the amp and the buck which cuts it to 5V to supply the rest. I'm not sure this is a good method to do this, I think I introduce lot of noise, also my circuit is not built super tight. 

I was really happy with my 3D printed circuit board frame, and my 3D printed button board replacement..

Not sure if there is a better 2.1 board I can get for this but, this was the only cheap/small enough option to fit in the case.

All that said it looks nice, is 2.1 audio, reuses the same buttons and sounds decent. 

![20240215_004433 (1)](https://github.com/sle118/squeezelite-esp32/assets/34561734/c519070d-13fb-4234-8150-66bb93620ed1)
![20240215_010007](https://github.com/sle118/squeezelite-esp32/assets/34561734/3f6a510e-d18d-4f6f-b156-07dad75387df)
